### ISSUE TITLE
[Task1&2-1] 数据集准备和数据抽样可视化检查

### ISSUE THREAD

**Issue by re-tourist** [re-tourist/sdn-for-super-resolution-reproduction#6]
**目标**
成功下载数据集 
实现一个脚本，随机抽样展示 HR / LR / 上采样 LR，验证数据对齐和缩放逻辑。

**输入**

- 当前数据读取逻辑
- 数据路径
- 目标尺寸和下采样倍率配置

**输出**

- 脚本：`scripts/inspect_dataset.py`
- 输出目录：`results/inspection/...`
- 至少若干组可视化图

**验收标准**

- 能随机导出多组样本图
- 每组图至少包含 HR、LR、upsampled LR
- 文件命名带 index
- 支持固定随机种子
- 图像范围和尺寸正确

**不允许顺手做**

- 不改模型代码
- 不加训练逻辑
- 不加评估逻辑

**Comment by re-tourist** [comment]
# Codex Prompt
你正在一个“复现 diffractive decoder 超分论文”的 Python 项目中工作。当前只做 Stage 1 / Stage 2 的基础任务之一：数据抽样可视化检查。

请你完成以下任务，并严格控制改动范围，不要顺手修改模型、训练、评估主流程。

【任务目标】
实现一个“数据抽样可视化脚本”，用于随机检查数据集中的 HR 图、LR 图及其上采样版本，验证：
1. LR 和 HR 是否正确对应同一样本
2. 下采样与上采样逻辑是否正确
3. 图像尺寸、值域、方向、归一化是否正常
4. 为后续 baseline 和 optical model 提供可信的数据检查入口

【你需要完成的具体内容】
1. 新增一个脚本，建议路径：
   scripts/inspect_dataset.py

2. 该脚本需要支持：
   - 从项目当前的数据集读取逻辑中加载样本
   - 随机抽取若干样本（默认 8 个）
   - 支持固定随机种子
   - 支持指定 split（如 train / val / test）
   - 支持指定输出目录
   - 支持指定上采样方式（至少 bicubic）
   - 支持指定可视化图像数量
   - 支持指定是否保存单张图和总览拼图

3. 每个样本至少可视化以下三项：
   - HR 图
   - LR 图
   - 将 LR 上采样回 HR 尺寸后的图（upsampled LR）
   如果容易实现，也可以额外显示：
   - 样本 index
   - 图像尺寸
   - min/max 值
   - 可选 difference map（先不强求）

4. 输出结果建议包括：
   - 每个样本单独一张图
   - 一张总览网格图（每行一个样本，三列：HR / LR / upsampled LR）
   - 一个简单的 summary 文本或 json，记录本次脚本运行参数：
     - split
     - num_samples
     - seed
     - output_dir
     - upsample_mode

5. 请优先复用项目现有的数据集读取代码，不要新造一套数据定义。
   如果现有数据接口不够直接，可以做最小封装，但不要改动训练主入口。

【实现要求】
1. 代码尽量清晰，结构简单
2. 不要引入新依赖，优先使用已有依赖（如 matplotlib、PIL、torch、torchvision）
3. 图像保存时要注意：
   - 灰度图显示正确
   - 值域统一，避免全黑全白
   - 标题里尽量标明 sample index 和 shape
4. 如果数据张量是 [C,H,W] 或 [H,W]，都要正确处理
5. 假设当前任务先以单通道灰度图为主，但代码最好不要写死到只能处理一种极端格式
6. 输出目录若不存在，要自动创建
7. 命令行参数建议使用 argparse

【明确限制】
- 不要修改模型文件
- 不要修改训练脚本
- 不要添加神经网络
- 不要实现评估指标
- 不要更改数据预处理主逻辑，除非为了兼容这个检查脚本必须做极小补丁，并在说明里明确写出
- 不要顺手做 baseline、PSNR、SSIM

【建议脚本接口】
尽量支持类似：
python scripts/inspect_dataset.py --split train --num-samples 8 --seed 42 --output-dir results/inspection/train --upsample bicubic

【你交付时请同时给出】
1. 修改/新增了哪些文件
2. 你对当前数据接口做了什么假设
3. 如何运行脚本
4. 输出会生成哪些文件
5. 有哪些潜在风险点或你不确定的地方

【验收标准】
只有当以下条件全部满足，任务才算完成：
1. 脚本可以独立运行
2. 能随机导出若干样本图
3. 每个样本至少包含 HR / LR / upsampled LR 三部分
4. 支持固定随机种子
5. 输出目录自动创建
6. 不影响现有训练/模型代码
7. 代码改动范围小且清晰

请先分析项目里可能已有的数据集读取方式，然后在最小改动原则下实现。


补充要求：
- 优先做成一个独立脚本，不要把逻辑塞进 train.py
- 优先读取现有 dataset / dataloader 配置，不要复制粘贴一套平行实现
- 如果项目当前还没有统一 dataset 工具函数，可以增加一个非常小的辅助函数，但不要重构整个数据模块
- matplotlib 输出要适合灰度图检查，避免默认 colormap 误导观察
- 如果遇到数据值域不确定，请在图标题或 summary 中额外打印 min/max


请按以下格式回复：
1. 任务理解
2. 计划修改的文件
3. 实现说明
4. 如何运行
5. 输出示例说明
6. 风险与待确认点
7. 完成状态（是否满足验收标准）
